app/components/candidate_interface/course_choices_review_component.rb

```diff
@@ -207,5 +206,17 @@ def has_multiple_sites?(application_choice)
     def has_multiple_courses?(application_choice)
       Course.current_cycle.where(provider: application_choice.provider).many?
     end
+
+    def application_choices
+      @application_choices ||= @application_form
+        .application_choices
+        .includes(:course, :site, :provider, :offered_course_option, :interviews)
+        .order(id: :asc)
+        .reject { |application_choice| ineligible_application_choice_states.include?(application_choice.status.to_sym) }
+    end
+
+    def ineligible_application_choice_states
+      ApplicationStateChange::SUCCESSFUL_STATES + ApplicationStateChange::UNSUCCESSFUL_END_STATES - %w[offer]
```

Comment:
Having looked at the constants again i think this should reject the application choices that aren't in the  `ACCEPTED_STATES` if an application choice in the accepted states is present.

When a candidate accepts an offer, they transition from the `offer` to `pending_conditions` state. At this point, all other application choices are rejected.

This means that is should never be possible to have more than 1 application choice in the `%i[pending_conditions conditions_not_met recruited offer_deferred]` states.

Reply:
I think this 
```

def application_choices
      @application_choices ||= @application_form
        .application_choices
        .includes(:course, :site, :provider, :offered_course_option, :interviews)
        .order(id: :asc)
        .reject { |application_choice| ineligible_application_choice_states.include?(application_choice.status.to_sym) }
    end

    def ineligible_application_choice_states
      ApplicationStateChange::SUCCESSFUL_STATES + ApplicationStateChange::UNSUCCESSFUL_END_STATES - %w[offer]
    end
```

should be something like this 

```
def application_choices
      @application_choices ||= reject_ineligible_application_choices
    end

    def reject_ineligible_application_choice_states
      if @application_form.application_choices.any? { |ac| ApplicationStateChange::ACCEPTED_STATES.include?(ac) }
        @application_form
         .application_choices
         .includes(:course, :site, :provider, :offered_course_option, :interviews)
         .order(id: :asc)
         .reject { |ac| ApplicationStateChange::ACCEPTED_STATES.exclude?(ac)}
       else
         @application_form
          .application_choices
          .includes(:course, :site, :provider, :offered_course_option, :interviews)
          .order(id: :asc)
       end
     end
```

This can def be cleaned up but you get the idea

Reply:
i think we can also add a method to the ApplicationChoice like 

```
def successful?
  ApplicationStateChange::ACCEPTED_STATES.include?(self)
end
```

then we can just do 


```

    def reject_ineligible_application_choice_states
      if @application_form.application_choices.any?(&:successful?)
        @application_form
         .application_choices
         .includes(:course, :site, :provider, :offered_course_option, :interviews)
         .order(id: :asc)
         .select(&:successful?)
       else
         @application_form
          .application_choices
          .includes(:course, :site, :provider, :offered_course_option, :interviews)
          .order(id: :asc)
       end
     end
```

Reply:
@davidgisbey hiya. I'm not quite following the ruby code, but your examples look good. Although in most cases, if there are 3 previous application choices at the bottom, the candidate will only see a single choice at the top, as you can only apply for 1 at a time during "apply again" at the moment.  The exception would be if the previous applications were from previous "cycles".

The intent here is mainly:

* To hide irrelevant application choices once an applicant has accepted one, so that they can focus on meeting those conditions (and be constantly reminded of any negative feedback)
* To give candidates who are applying again access to their previous application forms and feedback from providers (currently we don't link to these at all once you've submitted).

In both cases, only the minimum of information is needed, as they're unlikely to be looked at, and candidates can click through for the full detail.

Reply:
@davidgisbey @frankieroberto - thanks for clarifying. That all makes sense now 😁

Reply:
@TobyRet no probs. Let me know if you'd like me to test this at all, when it's ready. Cheers!